docs/executor_tutorial/google_lifesciences.rst

```diff
@@ -0,0 +1,880 @@
+that we have not customized this file from the basic tutorial to hard code 
+any storage or executor. We will be telling snakemake to change the executor 
+via command line arguments.
+
+.. code:: python
+
+    SAMPLES = ["A", "B"]
+
+    rule all:
+        input:
+            "plots/quals.svg"
+
+    rule bwa_map:
+        input:
+        fastq="samples/{sample}.fastq",
+        idx=multiext("genome.fa", ".amb", ".ann", ".bwt", ".pac", ".sa")
+    conda:
+        "environment.yml"
+    output:
+        "mapped_reads/{sample}.bam"
+    params:
+        idx=lambda w, input: os.path.splitext(input.idx[0])[0]
+    shell:
+        "bwa mem {params.idx} {input.fastq} | samtools view -Sb - > {output}"
+
+    rule bwa_map:
+        input:
+            "genome.fa",
+            "samples/{sample}.fastq"
+        conda:
+            "environment.yml"
+        output:
+            "mapped_reads/{sample}.bam"
+        shell:
+            "bwa mem {input} | samtools view -Sb - > {output}"
+
+    rule samtools_sort:
+        input:
+            "mapped_reads/{sample}.bam"
+        output:
+            "sorted_reads/{sample}.bam"
+        conda:
+            "environment.yml"
+        shell:
+            "samtools sort -T sorted_reads/{wildcards.sample} "
+            "-O bam {input} > {output}"
+
+    rule samtools_index:
+        input:
+            "sorted_reads/{sample}.bam"
+        output:
+            "sorted_reads/{sample}.bam.bai"
+        conda:
+            "environment.yml"
+        shell:
+            "samtools index {input}"
+
+    rule bcftools_call:
+        input:
+            fa="genome.fa",
+            bam=expand("sorted_reads/{sample}.bam", sample=SAMPLES),
+            bai=expand("sorted_reads/{sample}.bam.bai", sample=SAMPLES)
+        output:
+            "calls/all.vcf"
+        conda:
+            "environment.yml"
+        shell:
+            "samtools mpileup -g -f {input.fa} {input.bam} | "
+            "bcftools call -mv - > {output}"
+
+    rule plot_quals:
+        input:
+            "calls/all.vcf"
+        output:
+            "plots/quals.svg"
+        conda:
+            "environment.yml"
+        script:
+            "plot-quals.py"
+
+
+
+And let's also write the script in our present working directory for the last step
+to do the plotting - call this `plot-quals.py`:
+
+.. code:: python
+
+    import matplotlib
+    matplotlib.use("Agg")
+    import matplotlib.pyplot as plt
+    from pysam import VariantFile
+
+    quals = [record.qual for record in VariantFile(snakemake.input[0])]
+    plt.hist(quals)
+
+    plt.savefig(snakemake.output[0])
+
+
+Step 3: Run Snakemake
+:::::::::::::::::::::
+
+Now let's run Snakemake with the Google Life Sciences Executor.
+
+
+.. code:: console
+
+    snakemake --google-lifesciences --default-remote-prefix snakemake-testing-data --use-conda --google-lifesciences-region us-west1 --container-image snakemake/snakemake:v5.10.0
```

Comment:
Shouldn't the remote prefix match the bucket name where you uploaded the tutorial data above?

Reply:
The example for upload showed a generic example for uploading to a bucket with subfolder, e.g.,:

```python
export GOOGLE_APPLICATION_CREDENTIALS="/path/to/credentials.json"
python upload_google_storage.py my-snakemake-bucket/data  
```
but for the rest of the tutorial I used an actual bucket, snakemake-tutorial-data (without a subfolder). I chose the first generic to (very obviously say "this is a custom bucket I created and not something provided by snakemake). What I'll do is go back now and replace all instances of `snakemake-testing-data` with `my-snakemake-bucket/data` so to the user it's just clearly the same variable.

Reply:
Or the other option is to change just the first string - and actually I changed my mind and think we should do that, because I took screenshots of the bucket (and name) that wouldn't make sense.

Reply:
okay fixed locally, and I made it very clear. Will push soon.

Reply:
All set. I hope it's clearer now! I want the user to know that they can use a bucket root (without subfolder) or one with subfolder, and then have that be consistent to be the one I use in the following tutorial. I think this is good now.